### PR TITLE
chore(core): add ci workflows

### LINKED ISSUE #34: Set up CI for PR
Currently, PR is only tested in terms of whether app can be built.

To provide better developer experience for the community, it would be nice to add more CI so that anyone can ensure the quality of PRs without manual instruction.

---

- [ ] [semantic-prs](https://github.com/Ezard/semantic-prs) 
  - Commit messages in other OXC project follows conventional commits, if I'm correct
  - Since we squash-merge PRs, this action helps us keep the commit log of `main` branch clean, [if PR title is set as default PR commit message](https://github.blog/changelog/2022-05-11-default-to-pr-titles-for-squash-merge-commit-messages/).
- [ ] [labeler](https://github.com/actions/labeler)
  - OXC utilizes this action to add labels to the PR automatically
  - This could help for management of translation-related PR (see #13)
    - we could define `CODEOWNER` file if we have each language's maintainer, but if not, language labels can be a signal like "please review this PR if you are speaker of this language".

---

Any thoughts?

### PR DESCRIPTION
**Pull request by NozomuIkuta**
Resolves #34

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+lang:en:
+  - all:
+    - 'src/**/*.md'
+    - '!src/ja/**/*.md'
+
+lang:ja:
+  - all:
+    - '!src/**/*.md'
+    - 'src/ja/**/*.md'
```
Comment (NozomuIkuta) [inline .github/labeler.yml:14]: I defined 3 labels: 

- `core` for PRs and issues related to application (or, anything than docs itself)
- `lang:en` for English contents
- `lang:ja` for Japanese contents

Note that I defined just as a starting point of discussion.
I'm glad to hear your thoughts.

File: .github/workflows/ci-pr-label.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      contents: read
+      pull-requests: write
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+    steps:
+        - uses: actions/labeler@v4
+          with:
+            repo-token: ${{ secrets.GITHUB_TOKEN }}
+            sync-labels: true
+            dot: true
```
Comment (NozomuIkuta) [inline .github/workflows/ci-pr-label.yml:24]: This workflow behaves based on mapping or labels and paths defined in `.github/labeler.yml`.

**Comment by Boshen** [comment]
Oh wow! Thank you! I'll fix the required check.

File: .github/workflows/ci-semantic-pr.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+          types: |
+            docs
+            feat
+            fix
+            chore
+          scopes: |
+            core
+            en
+            ja
+          requireScope: true
```
Comment (NozomuIkuta) [inline .github/workflows/ci-semantic-pr.yml:30]: IMHO, 4 types and 3 scopes are enough for documentation site:

**Types**

- `docs` for any kinds of changes on documentation (e.g. fixing typos, new translation)
  - PRs whose commit message has `docs` type are supposed to be attached `lang:x` labels
- `feat` for application-level features (e.g. new page, new UI, etc.)
- `fix` for  application-level fixes
- `chore` for anything else (e.g. updating `README`, Change CI, etc.)

**Scopes**

- `core` for application
  - `docs(core)` would not be used in real operation
- `en` for English contents
- `ja` for Japanese contents

The reason why I defined both scopes and labels for i18n is that we can filter PRs and issues with labels, which is hard to achieve with scopes in commit massage only.

**Review by NozomuIkuta — COMMENTED** [review]
@Boshen 

Sorry, I created this PR without noticing you assigned yourself to the issue 🙏 
(And I'm surprised that you merged this while I writing this comment 😳)

Anyway, it's appreciated if you check the following comments that I left for further discussion.

cc: @Ubugeeei

**Comment by Ubugeeei** [comment]
@NozomuIkuta 
I think it's really good!

**Pull request merged**